enhancements/sig-architecture/26-policy-generator/README.md

```diff
@@ -0,0 +1,253 @@
+single one. This flexibility allows the workflow to reflect an organization’s standards and
+structure.
+
+### User Experience
+
+#### The Kustomization YAML File
+
+Kustomize is configured with a `kustomization.yaml` file. To use the generator plugin, a user must
+declare it under the generators section as shown in a `kustomization.yaml` file below:
+
+```yaml
+generators:
+  - policyGenerator.yaml
+patchesStrategicMerge:
+  - input/patch.yaml
+commonLabels:
+  custom: myApp
+```
+
+The `policyGenerator.yaml` file is the generator configuration file that Kustomize will provide. All
+other customizations such as "patches" and "commonLabels" apply on the generated policy returned
+from the new generator plugin.
+
+#### Generator Defaults
+
+By default, the generator plugin will fill in some default policy fields so that it’s easier for the
+user when they are getting started.
+
+The following policy fields will be required:
+
+- name
+- namespace
+
+The following policy fields will be optional:
+
+- annotations
+- policy.open-cluster-management.io/categories - defaults to "CM Configuration Management"
+- policy.open-cluster-management.io/controls - defaults to "CM-2 Baseline Configuration"
+- policy.open-cluster-management.io/standards - defaults to "NIST SP 800-53" (to be confirmed)
+- disabled - defaults to "false"
+- remediationAction - defaults to "inform"
+- severity - defaults to "low"
+
+#### The Generator Configuration File
+
+As shown above in the example `kustomization.yaml` file, a configuration file for the generator
+plugin was provided in the `policyGenerator.yaml` file. The following example explains what each
+field is used for:
+
+```yaml
+apiVersion: policy.open-cluster-management.io/v1
+kind: PolicyGenerator
+metadata:
+  # Required. Used to uniquely identify the configuration file.
+  name: ""
+
+# Required if multiple policies are consolidated in a PlacementBinding so that the generator can
+# create unique PlacementBinding names using the name given here.
+placementBindingDefaults:
+  # Set an explicit placement binding name to use rather than rely on the default.
+  name: ""
+
+# Required. Any default value listed here can be overridden under an entry in the policies array
+# except for "namespace".
+policyDefaults:
+  # Optional. Array of categories to be used in the policy.open-cluster-management.io/categories
+  # annotation. This defaults to ["CM Configuration Management"].
+  categories:
+    - "CM Configuration Management"
+  # Optional. Array of controls to be used in the policy.open-cluster-management.io/controls
+  # annotation. This defaults to ["CM-2 Baseline Configuration"].
+  controls:
+    - "CM-2 Baseline Configuration"
+  # Optional. This determines if a single configuration policy should be
+  # generated for all the manifests being wrapped in the policy.
+  # If set to false, a configuration policy per manifest will be generated.
+  # This defaults to true.
+  consolidateManifests: true
```

Comment:
will this and informKyvernoPolicies be orthogonal？

Reply:
Yes. `consolidateManifests` affects the structure of the policy:
```
Policy
  policyTemplates
    ConfigurationPolicy
      objectTemplate-1
      ...
      objectTemplate-n
```
vs
```
Policy
  policyTemplates
    ConfigurationPolicy-1
      objectTemplate-1
      ...
    ConfigurationPolicy-n
      objectTemplate-n
```